controllers/chat/getSignedChannelList.js

```diff
@@ -0,0 +1,73 @@
+const {responseSuccess} = require('../../utils/Responses');
+const GetstreamSingleton = require('../../vendor/getstream/singleton');
+
+const __queryChannelBuilder = async (userId, limit, offset) => {
+  console.log('querying channel', limit, offset);
+  const client = GetstreamSingleton.getChatInstance();
+  const channels = await client.queryChannels({members: {$in: [userId]}}, [{last_message_at: -1}], {
+    limit: limit,
+    offset: offset,
+    state: true
+  });
+  await client.disconnectUser();
+  return channels;
+};
+
+const getSignedChannelList = async (req, res) => {
+  const {userId} = req;
+  const {limit = 100, offset = 0} = req.query;
+
+  let totalFetched = 0;
+  const queriedChannels = [];
+  const promisedChannels = [];
+
+  const client = GetstreamSingleton.getChatInstance();
+  try {
+    await client.connectUser({id: req.userId}, req.token);
+
+    while (totalFetched < limit) {
+      const batchSize = Math.min(30, limit - totalFetched);
+      promisedChannels.push(
+        __queryChannelBuilder(userId, batchSize, Number(totalFetched) + Number(offset))
+      );
+      totalFetched += batchSize;
+      if (totalFetched >= limit) {
+        break;
+      }
+    }
+
+    const responses = await Promise.all(promisedChannels);
+    responses.forEach((response) => {
+      queriedChannels.push(...response);
+    });
+
+    await client.disconnectUser();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The while loop used to fetch channels in batches could potentially result in an infinite loop if there's an issue with the `__queryChannelBuilder` function that prevents it from fetching channels. Consider adding a safeguard to prevent this, such as a maximum number of iterations.

```diff
+    let iterationCount = 0;
     while (totalFetched < limit) {
+      if (iterationCount++ > MAX_ITERATIONS) {
+        throw new Error('Maximum number of iterations exceeded');
+      }
       const batchSize = Math.min(30, limit - totalFetched);
       promisedChannels.push(
         __queryChannelBuilder(userId, batchSize, Number(totalFetched) + Number(offset))
       );
       totalFetched += batchSize;
       if (totalFetched >= limit) {
         break;
       }
     }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->